src/controllers/actions_controller.js

```diff
@@ -42,7 +56,11 @@ ActionsController.post('/', function (request, response) {
  * Handle getting a list of actions.
  */
 ActionsController.get('/', function(request, response) {
-  response.status(200).json(Actions.getAll());
+  if (request.params.thingId) {
+    response.status(200).json(Actions.getByThing(request.params.thingId));
+  } else {
+    response.status(200).json(Actions.getAll());
```

Comment:
I propose one small change. I think we should start treating the gateway as a thing in its own right, so /actions is the action queue for the gateway thing, not a list of all actions requested by all things. Possibly the simplest way to implement this would be for /actions to return all actions with no thingId, rather than filtering by thingId. Does that make sense?

Reply:
That makes sense, I'll modify it accordingly. Thanks!